### PR TITLE
Add legacy flag for add-ons

### PR DESCRIPTION
**Pull request by nvdaes**
This has been [proposed on the add-ons mailing list](https://nvda-addons.groups.io/g/nvda-addons/message/21190)

### CODE REVIEW

**Comment by nvdaes** [comment]
I've performed integration tests with tox, not manual tests creating a json file, waiting for further comments.

File: _validate/addonVersion_schema.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
 			"sha256": "69D84CA8899800A5575CE31798293CD4FEBAB1D734A07C2E51E56A28E0DF8C82",
 			"sourceURL": "https://github.com/nvaccess/addon-datastore/",
 			"license": "GPL v2",
-			"licenseURL": "https://github.com/nvaccess/addon-datastore/license.MD"
+			"licenseURL": "https://github.com/nvaccess/addon-datastore/license.MD",
+			"legacy": true
```
Comment (seanbudd) [inline _validate/addonVersion_schema.json]: lets undo this and leave legacy out of the sample/example

File: _validate/createJson.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ def generateJsonFile(
 		url: str,
 		licenseName: str,
 		licenseUrl: typing.Optional[str],
+		legacy: typing.Optional[bool],
```
Comment (seanbudd) [inline _validate/createJson.py]: I don't think legacy should be included in create JSON - I don't think we should allow or encourage more legacy submissions to be created.

**Review by seanbudd — COMMENTED** [review]
Apologies for the late review - this was missed.

**Comment by nvdaes** [comment]
Hi @seanbudd : I've addressed your review. It's true: new submissions shouldn't be created as legacy.

File: _validate/createJson.py [hunk 3]
```diff
@@ -108,7 +108,6 @@ def _createDictMatchingJsonSchema(
 		addonData["homepage"] = homepage
 	if licenseUrl:
 		addonData["licenseURL"] = licenseUrl
-
```
Comment (seanbudd) [inline _validate/createJson.py]: this diff should be removed

File: _tests/test_createJson.py [hunk 4]
```diff
@@ -61,6 +61,7 @@ def test_contentsMatchesExampleFile(self):
 			url="https://github.com/nvaccess/dont/use/this/address/fake.nvda-addon",
 			licenseName="GPL v2",
 			licenseUrl="https://www.gnu.org/licenses/gpl-2.0.html",
+			legacy=False,
```
Comment (seanbudd) [inline _tests/test_createJson.py]: this diff should be removed

**Comment by nvdaes** [comment]
I've addressed your comments and checks pass.

**Review by seanbudd — APPROVED** [review]
Thanks @nvdaes

**Pull request merged**